clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -144,6 +145,23 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
     return mlir::cir::TypeInfoAttr::get(anonStruct.getType(), anonStruct);
   }
 
+  mlir::TypedAttr getZeroInitAttr(mlir::Type ty) {
+    if (ty.isa<mlir::cir::IntType>())
+      return mlir::cir::IntAttr::get(ty, 0);
+    if (ty.isa<mlir::FloatType>())
+      return mlir::FloatAttr::get(ty, 0.0);
+    if (auto arrTy = ty.dyn_cast<mlir::cir::ArrayType>()) {
+      // FIXME(cir): We should have a proper zero initializer CIR instead of
+      // manually pumping zeros into the array.
```

Comment:
In fact both `cir.const_struct` and `cir.const_array` need to support `#cir.zero` (just filed #129), can you please assert on a `UnimplementedFeature::` here too?